### ISSUE TITLE
Process directories in repo root

### ISSUE THREAD

**Issue by stdavids** [black-parrot-examples/bsg_scratch_designs#1]
Inside each of the designs is a tcl/hard/<process> directory which is supposed to be how designs express differences from processes. What are the benefits of also having the full design separated at the root of the repository as well?

**Comment by dpetrisko** [comment]
Yeah, that’s a good point. I did it this way just because they were already separated this way, but I think your proposal makes more sense

**Comment by dpetrisko** [comment]
One benefit would be if the designs change by generation — just as we have separate repos per tapeout, the RTL might change per process node. But by virtue of these being “scratch designs”, it might be okay to break previous designs as we work on current designs

**Comment by stdavids** [comment]
It's an interesting point about RTL changes that process specific. Naively one would want to say that RTL is totally process agnostic but in reality we know that is not totally true for well optimized designs. I think for the most part we try and abstract those differences in basejump STL. It is fairly rare for the RTL inside of a design to be process specific.

**Comment by dpetrisko** [comment]
Yeah, I agree. My point is less that the RTL changes actually because of the process node and more that when the RTL changes, it breaks TCL scripts because of renaming, different mems, etc. 

Since we only typically work on a process node at a time, what will probably end up happening is a scratch design will have 1 functional TCL/hard folder and several broken ones.  But I think this is fine because it’s a scratch design, so needing to be slightly touched up is acceptable. 

One idea to fix the problem is to have different commit tags per process in the Makefile.
